mlflow/entities/model_registry/prompt_version.py

```diff
@@ -19,17 +19,133 @@
 
 # Alias type
 PromptVersionTag = ModelVersionTag
 
 
+class PromptModelConfig(BaseModel):
+    """
+    Configuration for a model associated with a prompt, including model name and inference
+    parameters.
+    This class provides a structured way to store model-specific settings alongside prompts,
+    ensuring reproducibility and clarity about which model and parameters were used with a
+    particular prompt version.
+
+    Args:
+        model_name: The name or identifier of the model (e.g., "gpt-4", "claude-3-opus").
+        temperature: Sampling temperature for controlling randomness (typically 0.0-2.0).
+            Lower values make output more deterministic, higher values more random.
+        max_tokens: Maximum number of tokens to generate in the response.
+        top_p: Nucleus sampling parameter (typically 0.0-1.0). The model considers tokens
+            with top_p cumulative probability mass.
+        top_k: Top-k sampling parameter. The model considers only the k most likely tokens.
+        frequency_penalty: Penalty for token frequency (typically -2.0 to 2.0). Positive
+            values reduce repetition of tokens based on their frequency in the text so far.
+        presence_penalty: Penalty for token presence (typically -2.0 to 2.0). Positive
+            values increase likelihood of introducing new topics.
+        stop_sequences: List of sequences that will cause the model to stop generating.
+        extra_params: Additional model-specific parameters not covered by the standard fields.
+            This allows for flexibility with provider-specific or experimental parameters.
+
+    Example:
+
+    .. code-block:: python
+
+        from mlflow.entities.model_registry import PromptModelConfig
+
+        # Basic configuration
+        config = PromptModelConfig(
+            model_name="gpt-4",
+            temperature=0.7,
+            max_tokens=1000,
+        )
+        # Configuration with extra provider-specific params
+        config = PromptModelConfig(
+            model_name="claude-3-opus",
+            temperature=0.5,
+            max_tokens=2000,
+            extra_params={
+                "anthropic_version": "2023-06-01",
+                "response_metadata": {"cache_control": True},
+            },
+        )
+        # Use with prompt registration
+        import mlflow
+
+        mlflow.genai.register_prompt(
+            name="my_prompt",
+            template="Analyze this: {{text}}",
+            model_config=config,
+        )
+    """
+
+    model_name: str | None = None
+    temperature: float | None = Field(None, ge=0)
+    max_tokens: int | None = Field(None, gt=0)
+    top_p: float | None = Field(None, ge=0, le=1)
+    top_k: int | None = Field(None, gt=0)
+    frequency_penalty: float | None = None
+    presence_penalty: float | None = None
+    stop_sequences: list[str] | None = None
+    extra_params: dict[str, Any] = Field(default_factory=dict)
```

Comment:
can `extra_params` contain the known fields?

Reply:
yes good question - it's allowed and would work fine, but not the good practice because it can break the UI rendering or so.